sdk/core/azure_core_amqp/src/session.rs

```diff
@@ -144,8 +144,8 @@ pub mod builders {
         }
         pub fn with_properties<K, V>(mut self, properties: impl Into<AmqpOrderedMap<K, V>>) -> Self
         where
-            K: Into<AmqpSymbol> + PartialEq + Default + Debug,
```

Comment:
Why is `Debug` actually needed here? Typically only include constraints needed in the impl. Seems nothing should require a `Debug` impl, or does collecting into an `AmqpOrderedMap` actually?

Reply:
It's not, removed.